app/models/ordergroup.rb

```diff
@@ -148,6 +148,27 @@ def account_updated
     financial_transactions.last.try(:created_on) || created_on
   end
 
+  def self.sort_by_param(param)
+    sort_param_map = {
+      "name" => "name",
+      "name_reverse" => "name DESC",
+      "members_count" => "count(*)",
+      "members_count_reverse" => "count(*) DESC",
+      "last_user_activity" => "max(users.last_activity)",
+      "last_user_activity_reverse" => "max(users.last_activity) DESC",
+      "last_order" => "max(orders.starts)",
+      "last_order_reverse" => "max(orders.starts) DESC"
+    }
+
+    @ordergroup = case param
+                  when "members_count", "members_count_reverse", "last_user_activity", "last_user_activity_reverse" then @ordergroup.left_joins(:users).group("groups.id")
+                  when "last_order", "last_order_reverse" then @ordergroup.left_joins(:orders).group("groups.id")
+                  else
+                    Ordergroup
+                  end
+    @ordergroup.order(sort_param_map[param])
+  end
+
```

Comment:
what do you think about writing it the following way?

```ruby
ret = self
ret = ret.left_joins(:users).group("groups.id") if param.starts_with?("members_count", "last_user_activity")
ret = ret.left_joins(:orders).group("groups.id") if param.starts_with?("last_order")
ret.order(sort_param_map[param])
```

Reply:
I like it. It's much more readable. I didn't know about these expression modifiers.

Reply:
If it's okay I would name the variable `result`, as I like to avoid abbreviations. But as naming is rather subjective, I am fine with either way, especially if `ret` is commonly used in the project?

Reply:
you'll find both in the code, so just go with what you like more